### PR TITLE
fix: Hough Transform first implementation clang-tidy

### PR DESCRIPTION
**Pull request by andiwand**
based on https://github.com/acts-project/acts/pull/1305

clang tidy fixes

### CODE REVIEW

File: Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+                houghHist(y, x).second.size()) {
+              return false;
+            }
+            if (houghHist(y + j, x + i).second.size() ==
+                    houghHist(y, x).second.size() &&
+                j <= 0 && i <= 0) {
+              return false;  // favor bottom-left (low phi, low neg q/pt)
+            }
+          }
+        }
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp:274]: Do you think combining the according `if`s in `else if`s could make this section more readable?

```suggestion
        if (y + j < houghHist.size(0) && x + i < houghHist.size(1)) {
          if (houghHist(y + j, x + i).first > houghHist(y, x).first) {
            return false;
          } else if (houghHist(y + j, x + i).first == houghHist(y, x).first) {
            if (houghHist(y + j, x + i).second.size() >
                houghHist(y, x).second.size()) {
              return false;
            } else if (houghHist(y + j, x + i).second.size() ==
                       houghHist(y, x).second.size()) {
              if (j <= 0 && i <= 0) {
                return false;  // favor bottom-left (low phi, low neg q/pt)
              }
            }
          }
        }
```

File: Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+    std::copy(v.begin(), v.end() - 1, std::ostream_iterator<T>(oss, ", "));
+    oss << v.back();
+  }
+  oss << "]";
+  return oss.str();
+}
+
+double ActsExamples::HoughTransformSeeder::yToX(double y, double r,
+                                                double phi) const {
+  double d0 = 0;  // d0 correction TO DO allow for this
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp:311]:
```suggestion
  double d0 = 0;  // d0 correction TODO: allow for this
```

File: Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+          for (auto measurementIndex : m_houghHist(y, x).second) {
+            HoughMeasurementStruct* meas =
+                houghMeasurementStructs[measurementIndex].get();
+            hitIndicesAll[meas->layer].push_back(meas->indices);
+            nHitsPerLayer[meas->layer]++;
+          }
+          std::vector<std::vector<int>> combs = getComboIndices(nHitsPerLayer);
+
+          for (auto [icomb, hit_indices] :
+               Acts::enumerate(combs)) {  // loop over all the combination
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp:156]:
```suggestion
               Acts::enumerate(combs)) {  // loop over all the combinations
```

**Review by AJPfleger — COMMENTED** [review]
In addition to my other comments: I am not quite sure if everything with the `///`-doxygen comments should have them and vice versa. Maybe if you have time, you could have a look over them again.

**Comment by andiwand** [comment]
thanks for your comments @AJPfleger but I fear they should go here https://github.com/acts-project/acts/pull/1305

my PR is only there to fix the clang tidy issues

@asalzburger this seems to pass so you could merge the original PR and this one on top to fix the CI

**Review by asalzburger — APPROVED** [review]
Sounds good, I included Alexander's suggestion.

File: Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp:274]: Let's leave it like this, for the moment.

**Comment by asalzburger** [comment]
Updating the branch with main.

**Comment by asalzburger** [comment]
Meh, that was stupid of me, I have merged Alexander's suggestions into that ...

**Comment by asalzburger** [comment]
@andiwand - I think best is we make a cherrypick version with only the clang-tidy commit.

**Comment by andiwand** [comment]
@asalzburger I reverted to my original commit so this should be mergable again

**Comment by asalzburger** [comment]
Thanks!

**Comment by asalzburger** [comment]
Hi @andiwand - the original Hough Transform PR is merged, but I see some conflicts now, can you try to resolve?

**Comment by andiwand** [comment]
@asalzburger conflicts should be gone now :+1:

**Pull request merged**